python/pyiceberg/catalog/rest.py

```diff
@@ -335,6 +349,53 @@ def _handle_non_200_response(self, exc: HTTPError, error_handler: Dict[int, Type
 
         raise exception(response) from exc
 
+    def _init_sigv4(self) -> None:
+        from urllib import parse
+
+        import boto3
+        from botocore.auth import SigV4Auth
+        from botocore.awsrequest import AWSRequest
+        from requests import PreparedRequest
+        from requests.adapters import HTTPAdapter
+
+        class SigV4Adapter(HTTPAdapter):
+            def __init__(self, **properties: str):
+                super().__init__()
+                self._properties = properties
+
+            def add_headers(self, request: PreparedRequest, **kwargs: Any) -> None:  # pylint: disable=W0613
+                session = boto3.Session()
+                credentials = session.get_credentials()
+                creds = credentials.get_frozen_credentials()
+                region = self._properties.get(SIGV4_REGION, session.region_name)
+                service = self._properties.get(SIGV4_SERVICE, "execute-api")
+
+                url = str(request.url).split("?")[0]
+                query: str = str(parse.urlsplit(request.url).query)
+                params = dict(parse.parse_qsl(query))
+
+                # remove the connection header as it will be updated after signing
+                del request.headers["connection"]
+
+                aws_request = AWSRequest(
+                    method=request.method, url=url, params=params, data=request.body, headers=dict(request.headers)
+                )
+
+                SigV4Auth(creds, service, region).add_auth(aws_request)
+                original_header = request.headers
+                signed_headers = aws_request.headers
+                relocated_headers = {}
+
+                # relocate headers if there is a conflict with signed headers
+                for header, value in original_header.items():
+                    if header in signed_headers and signed_headers[header] != value:
+                        relocated_headers[f"Original-{header}"] = value
+
+                request.headers.update(relocated_headers)
+                request.headers.update(signed_headers)
+
+        self._session.mount(self.uri, SigV4Adapter())
```

Comment:
The properties are not passed in.
```suggestion
        self._session.mount(self.uri, SigV4Adapter(**self.properties))
```
Otherwise you would never be able to set the region and service.